spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowServerConfiguration.java

```diff
@@ -307,6 +302,18 @@ public TaskRepositoryInitializer taskRepositoryInitializerForDefaultDB(DataSourc
 			return taskRepositoryInitializer;
 		}
 
+		@Bean
+		public DefinitionRepositoryInitializer definitionRepositoryInitializerForDefaultDB(DataSource dataSource, Server server) {
```

Comment:
Not sure if this would help.  The issue is that the server and the datasource both must be up before  this bean can be created (and the database initialized). By stating that the sever autowiring is not required, the following scenario could occur when connecting to the H2:
* Datasource is available.  Server is not available.   The definitionRepositoryInitializerForDefaultDB gets called and the initialization fails because datasource failed to connect to the h2 server.